doc/source/reference/random/index.rst

```diff
@@ -0,0 +1,237 @@
+.. _numpyrandom:
+
+.. currentmodule:: numpy.random
+
+numpy.random
+============
+
+Numpy's random number routines produce psuedo random numbers using
+combinations of generators and distributions:
+
+* Generators: Objects that generate random numbers. These are typically
+  unsigned integer words filled with sequences of either 32 or 64 random bits.
+* Distributions: Objects that transform sequences of random bits from a
+  generator into sequences of numbers that follow a specific probability
+  distribution (such as uniform, Normal or Binomial) within a specified
+  interval.
+
+Since Numpy version 1.17.0 the `RandomGenerator` can be initialized with a
+number of different Random Number Generators (RNG)s, and exposes many
+different probability distributions. See
+`NEP 19 <https://www.numpy.org/neps/nep-0019-rng-policy.html>`_ for context on
+the updated random Numpy number routines. The legacy `mtrand.RandomState`
+random number routines are still available, but limited to a single generator.
+
+For convenience and backward compatibility, a single `mtrand.RandomState`
+instance's methods are imported into the numpy.random namespace, see `legacy`
+for the complete list.
+
+Quick Start
+-----------
+
+By default, `RandomGenerator` uses normals provided by
+`xoroshiro128.Xoroshiro128` which will be faster than the legacy methods in
+`mtrand.RandomState`
+
+.. code-block:: python
+
+  # As replacement for numpy.random.RandomState
+  from numpy import random
+  random.standard_normal()
+
+`RandomGenerator` can be used as a direct replacement for
+`~RandomState`, although the random values are generated by
+`~xoroshiro128.Xoroshiro128`. The `RandomGenerator` holds an instance of a RNG.
+It is accessable as ``gen.brng``.
+
+.. code-block:: python
+
+  # As replacement for RandomState()
+  from numpy.random import RandomGenerator
+  rg = RandomGenerator()
+  rg.standard_normal()
+
+
+Seeds can be passed to any of the basic RNGs. Here `mt19937.MT19937` is used
+and the ``RandomGenerator`` is accessed via the attribute `mt19937.MT19937.
+generator`.
+
+.. code-block:: python
+
+  from numpy.random import MT19937
+  rg = MT19937(12345).generator
+  rg.standard_normal()
+
```

Comment:
@bashtage If one wanted to write this code a bit more expanded would the below variable names be good:

```
from numpy.random import MT19937

random_generator_container = MT19937(12345)
random_generator = random_generator_container.generator

print(random_generator.standard_normal())
```

I'm asking for my own understanding, but do you think it could be useful to the reader to have the container concept explained?